spinta/config/__init__.py

```diff
@@ -74,115 +73,106 @@
     'debug': False,
 }
 
-TEMPLATES = {
-    ('backends',): 'default',
-    ('manifests',): 'default',
-}
 
+class Config:
+
+    def __init__(self):
+        self._config = {
+            'cliargs': {},
+            'environ': {},
+            'envfile': {},
+            'default': {},
+        }
+        self.add(CONFIG)
+
+    def add(self, config):
+        self._config['default'].update(_traverse(config))
+        self._config['default'].update(_get_inner_keys(self._config['default']))
+
+    def add_cli_args(self, args):
+        for arg in args:
+            name, value = arg.split('=', 1)
+            key = tuple(name.split('.'))
+            self._config['cliargs'][key] = value
+        self._config['cliargs'].update(_get_inner_keys(self._config['cliargs']))
+
+    def add_env_file(self, path):
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+
+        if not path.exists():
+            return
+
+        with path.open() as f:
+            for line in f:
+                if line.startswith('#'):
+                    continue
+                line = line.strip()
+                if line == '':
+                    continue
+                if '=' not in line:
+                    continue
+                name, value = line.split('=', 1)
+                self._config['envfile'][name] = value
+
+    def set_env(self, environ):
+        self._config['environ'] = environ
+
+    def get(self, *key, default=None, cast=None, env=True, required=False):
+        value = self._get_config_value(key, default, env)
+
+        if cast is not None:
+            if cast is list and isinstance(value, str):
+                value = value.split(',')
```

Comment:
Hardcoded case when we want to cast string as list.

What if we'll want to split on `.`, `:`, `;`? Or just just convert a string to list `list(value_str)` ?

It requires developers to know implementation details of how `Config.get` works.

Reply:
This should be explained in documentation. Here, in order to pass a list via environment variables, you need to separate them with commas. That just a requirement.